stats/json/collector.go

```diff
@@ -94,44 +120,48 @@ func (c *Collector) HandleMetric(m *stats.Metric) {
 	}
 
 	c.seenMetrics = append(c.seenMetrics, m.Name)
-	env := WrapMetric(m)
-	row, err := json.Marshal(env)
+	err := c.encoder.Encode(WrapMetric(m))
 
-	if env == nil || err != nil {
-		logrus.WithField("filename", c.fname).Warning(
+	if err != nil {
+		logrus.WithField("filename", c.fname).WithField("error", err).Warning(
 			"JSON: Envelope is nil or Metric couldn't be marshalled to JSON")
 		return
 	}
-
-	row = append(row, '\n')
-	_, err = c.outfile.Write(row)
-	if err != nil {
-		logrus.WithField("filename", c.fname).Error("JSON: Error writing to file")
-	}
 }
 
 func (c *Collector) Collect(scs []stats.SampleContainer) {
+	c.bufferLock.Lock()
+	defer c.bufferLock.Unlock()
 	for _, sc := range scs {
+		c.buffer = append(c.buffer, sc.GetSamples()...)
+	}
+}
+
+func (c *Collector) commit() {
+	c.bufferLock.Lock()
+	samples := c.buffer
+	c.buffer = nil
+	c.bufferLock.Unlock()
+	logrus.WithField("filename", c.fname).Debug("JSON: Writing JSON metrics")
+	var start = time.Now()
+	var count int
+	for _, sc := range samples {
+		var samples = sc.GetSamples()
+		count += len(samples)
 		for _, sample := range sc.GetSamples() {
+			sample := sample
 			c.HandleMetric(sample.Metric)
-
-			env := WrapSample(&sample)
-			row, err := json.Marshal(env)
-
-			if err != nil || env == nil {
-				// Skip metric if it can't be made into JSON or envelope is null.
-				logrus.WithField("filename", c.fname).Warning(
-					"JSON: Envelope is nil or Sample couldn't be marshalled to JSON")
-				continue
-			}
-			row = append(row, '\n')
-			_, err = c.outfile.Write(row)
+			err := c.encoder.Encode(WrapSample(&sample))
 			if err != nil {
-				logrus.WithField("filename", c.fname).Error("JSON: Error writing to file")
+				// Skip metric if it can't be made into JSON or envelope is null.
+				logrus.WithField("filename", c.fname).WithField("error", err).Warning(
+					"JSON: Sample couldn't be marshalled to JSON")
 				continue
 			}
 		}
 	}
+	logrus.WithField("filename", c.fname).WithField("t", time.Since(start)).
```

Comment:
Maybe change `"t"` to `"start"` will give more meaningful log.

Reply:
but it's not `start` as far as I can see, it's the time from start, which is often denoted with `t`, so this LGTM